### PR TITLE
Backport 2.7: Fix dead code in mbedtls_mpi_exp_mod()

### PR DESCRIPTION
**Pull request by k-stachowiak**
This is a backport of #2317

### CODE REVIEW

**Comment by Patater** [comment]
Please rebase to update for all.sh changes.

**Comment by k-stachowiak** [comment]
@Patater I have rebased the PR.
@RonEld, @AndrzejKurek could you please re-review?

File: ChangeLog [hunk 1]
```diff
@@ -21,6 +21,10 @@ Changes
      produced by some optimizing compilers, showing up as failures in
      e.g. RSA or ECC signature operations. Reported in #1722, fix suggested
      by Aurelien Jarno and submitted by Jeffrey Martin.
+   * Remove dead code from bignum.c in the default configuration.
+     Found by Coverity, reported and fixed by Peter Kolbus. Fixes #2309.
```
Comment (pkolbus) [inline ChangeLog]: Please credit my employer as well for both of these: "Peter Kolbus (Garmin)"

Comment (k-stachowiak) [reply ChangeLog]: Done. Thanks for a quick response!

**Comment by k-stachowiak** [comment]
@AndrzejKurek @RonEld Could you please review the change log update?

**Comment by pkolbus** [comment]
LGTM, thanks

**Pull request merged**